docs/rest.md

```diff
@@ -0,0 +1,622 @@
+string. For example, `"1234567891234567"` represents
+`2009-02-13T23:31:31.234567`.
+
+## /
+### GET
+
+* Description: List of supported APIs
+* Authorization: guest
+* Operation: sync
+* Return: list of supported API endpoint URLs (by default `["/1.0"]`)
+
+
+## /1.0
+### GET
+
+* Description: Server configuration and environment information
+* Authorization: guest
+* Operation: sync
+* Return: Dict with the operating system’s key values.
+
+#### Sample result:
+
+```javascript
+{
+ "default_channel": "edge",
+ "flavor": "core",
+ "api_compat": "0",           // increased on minor API changes
+ "release": "15.04",
+ "store": "store-id"          // only if not default
+}
+```
+
+## /1.0/packages
+### GET
+
+* Description: List of packages
+* Authorization: trusted
+* Operation: sync
+* Return: list of URLs for packages this Ubuntu Core system can handle.
+
+The result is a JSON object with a packages key; its value is itself a
+JSON object whose keys are qualified package names (e.g.,
+hello-world.canonical), and whose values describe that package.
+
+Sample result:
+
+```javascript
+{
+ "packages": {
+   "dd.canonical": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "/icons/dd.png",
+     "installed_size": "-1",          // always -1 if not installed
+     "name": "dd",
+     "origin": "canonical",
+     "resource": "/1.0/packages/dd.canonical",
+     "status": "not installed",
+     "type": "app",
+     "vendor": "Somebody",
+     "version": "0.1"
+   },
+   "pastebinit.mvo": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "http://storeurl/icon.png",
+     "name": "pastebinit.mvo",
+     "installed_size": "-1",
+     "operation": {
+       "created_at": 1415639996,
+       "may_cancel": true,
+       "resource": "/1.0/packages/pastebinit.mvo",
+       "status": "Running",
+       "status_code": 100,
+       "updated_at": 1415639996
+     },
+     "origin": "mvo",
+     "resource": "/1.0/packages/pastebinit.mvo",
+     "rollback_available": "0.9",
+     "status": "installing",
+     "type": "app",
+     "update_available": "1.1",
+     "vendor": "Michael Vogt",
+     "version": "0.8"
+   },
+   "ubuntu-core.canonical": {
+     "description": "A description",
+     "download_size": "23456",
+     "icon": "",               // core might not have an icon
+     "installed_size": "-1",   // core doesn’t have installed_size (yet)
+     "name": "ubuntu-core",
+     "origin": "canonical",
+     "resource": "/1.0/packages/ubuntu-core.canonical",
+     "status": "active",
+     "type": "core",
+     "vendor": "Canonical",
+     "version": "43"
+   }
+ }
+}
+```
+
+#### Fields
+* `packages`
+    * `status`: can be either `not installed`, `installing`, `installed`,
+      `uninstalling`, `active` (i.e. is current), `removed` (but data present),
+      `purging`. For statuses that signal a background operation is in course,
```

Comment:
`installing`, `uninstalling` and `purging` are transient states. There is no `purged` state, because a purged package is undistinguishable from a non-installed package.